### ISSUE TITLE
Tree View lacks accessible names in 2.0.3

### ISSUE THREAD

**Issue by bdryanovski** [clr-team/planning#75]
## Description of work

The treenode buttons lack accessible names like the one below lack accessible names. Screen reader users will not understand the purpose of these buttons when they read them.

The Tree Views lack accessible names. Screen reader users will not know the purpose of the tree controls.

This ticket should be for only 2.0 implementation.

## Issue(s)

* https://github.com/vmware/clarity/issues/3570

## Reviewers

* Jeeyun
* Jeremy

## Definition of done

* [ ] Tests completed
* [ ] Reviewed by peers
* [ ] Design sign off (if necessary)
* [ ] Merged into master/topic branch
* [ ] Backport (if necessary)
